src/RegistrationController.js

```diff
@@ -61,86 +65,143 @@ function (
   return BaseLoginController.extend({
     className: 'registration',
     initialize: function() {
-      var clientId = this.options.settings.get('clientId');
-      var registrationApi = this.options.settings.get('baseUrl')+'/api/v1/registration/'+clientId;
-
+      var self = this;
+      // setup schema
       var Schema = RegistrationSchema.extend({
-        url: registrationApi+'/form'
+        settings: this.options.settings,
+        url: this.getRegistrationApiUrl()+'/form'
       });
-      
       var schema = new Schema();
-
-      var createRegistrationModel = _.bind(function(modelProperties){
-        var Model = Okta.Model.extend({
-          url: registrationApi+'/register',
-          settings: this.settings,
-          appState: this.options.appState,
-          props: modelProperties,
-          local: {
-            activationToken: 'string'
-          },
-          toJSON: function() {
-            var data = Okta.Model.prototype.toJSON.apply(this, arguments);
-            return {userProfile: data};
-          },
-          parse: function(resp) {
-            this.set('activationToken', resp.activationToken);
-            delete resp.activationToken;
-            return resp;
-          }
-        });
-        return new Model();
-      }, this);
-
-      var postSchemaFetch = _.bind(function(){
-        var properties = schema.properties;
-        var modelProperties = properties.createModelProperties();
-
-        this.model = createRegistrationModel(modelProperties);
-
-        var form = new Form(this.toJSON());
-
-        this.listenTo(form, 'saved', _.bind(function() {
-          var activationToken = this.get('activationToken');
-          if (activationToken) {
-            var authClient = this.appState.settings.authClient;
-            authClient.signIn({
-              token: activationToken
-            })
-            .then(_.bind(function(transaction) {
-              RouterUtil.routeAfterAuthStatusChange(this, null, transaction.data);
-            }, this));
-          } else {
-            this.appState.set('username', this.get('userName'));
-            this.appState.trigger('navigate', 'signin/register-complete');
-          }
-        }, this.model));
-
-        properties.each(function(schemaProperty) {
-          var inputOptions = RegistrationFormFactory.createInputOptions(schemaProperty);
-          var subSchemas = schemaProperty.get('subSchemas');
-          var name = schemaProperty.get('name');
-          form.addInput(inputOptions);
-          if (name === 'password' && subSchemas) {
-            form.add(SubSchema.extend({id: 'subschemas-' + name, subSchemas: subSchemas}));
-          }
-        });
+      // register parse complete event listener
+      schema.on('parseComplete', function(updatedSchema) {
```

Comment:
Looking much better with the fetchInitialData, but can we add this part in there as well? That way, we're getting all the async stuff done up front and won't have to deal with it in a callback here.

Reply:
I had the parseComplete listener inside fetchInitialData, but as discussed on IM had some trouble wiring up the tests. I'll give it another shot after fixing the broken tests.

Reply:
@rchild-okta this is now addressed with my latest commit.